### PR TITLE
Clarify Revise and testing workflow

### PR DESCRIPTION
**Pull request by goretkin**
This didn't make sense to me when I first read it, so I'm trying to make an improvement. xref: https://github.com/JuliaLang/julia/issues/35596

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -86,6 +86,7 @@ Julia's documentation source files are stored in the `doc/` directory and all do
 make docs
 ```
 
+
```
Comment (StefanKarpinski) [inline CONTRIBUTING.md:89]:
```suggestion
```

**Comment by StefanKarpinski** [comment]
cc @Keno since you added the revise/test stuff.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -242,9 +243,9 @@ If you need to restart your Julia session, just start at step 2 above.
 built and incorporate them automatically. You only need to rebuild
 Julia if you made code-changes that Revise cannot handle.
 
-For convenience, there are also `test-revise-*` targets for every `test-*`
-target that use Revise to load any modifications to Base into the current
-process before running the corresponding test. This can be useful as a shortcut
+For convenience, there are also `test-revise-*` targets for every [`test-*`
+target](https://github.com/JuliaLang/julia/blob/master/CONTRIBUTING.md#running-specific-tests) that use Revise to load any modifications to Base into the current
+system image before running the corresponding test. This can be useful as a shortcut
```
Comment (timholy) [inline CONTRIBUTING.md:248]: This makes it sound like the `*.so` file on disk is modified.

Comment (goretkin) [reply CONTRIBUTING.md:248]: Good point. I was looking for something more specific than "process", but maybe that's the right word to begin with.

The system image is the serialization of ______ to disk. The state of the julia runtime?

**Comment by ViralBShah** [comment]
I'm merging since this feels like an improvement, but if we should be rewording, please chime in and I can update it in a separate PR.

**Pull request merged**